lib/iex/lib/iex/autocomplete.ex

```diff
@@ -37,6 +37,26 @@ defmodule IEx.Autocomplete do
     end
   end
 
+  defp get_helper(expr) do
```

Comment:
I think a faster implementation would be this:

```elixir
defp get_helper(expr) do
  case Enum.reverse(expr) do
    [helper, space_or_paren, code | _]
    when helper in 'bt' and
           space_or_paren in ' (' and
           (char in ?A..?Z or char in ?a..?z or char in ?0..?9 or char in '_:') ->
      helper #=> returns ?b or ?t

    _ ->
      nil
  end
end
```

Can you please try it out? Also, please run `mix format` at the root of the project, as that will fix the build. :)

Reply:
Ah cool,  although the only thing with this implementation is that it won't catch the case where there are more than 1 space e.g. `t  Enum` or if the helpers are more than just a single character. If this is acceptable though, then this is definitely a much better implementation.

I've tried it out and all the tests are passing.

Reply:
@jgmchan oh, I see, so we can add a test case for multiple spaces. Then this implementation should be enough:

```elixir
defp get_helper(expr) do
  with [helper | rest] when helper in 'bt' <- Enum.reverse(expr),
       [space_or_paren, char | _] <- squeeze_spaces(rest),
       true <- space_or_paren in ' (' and
                 (char in ?A..?Z or char in ?a..?z or char in ?0..?9 or char in '_:') do
    helper
  else
    _ ->
      nil
  end
end

defp squeeze_spaces('  ' ++ rest), do: squeeze_spaces([?\s | rest])
defp squeeze_spaces(rest), do: rest
```

Reply:
Sweet, I've added the extra tests and used the above implementation. I've also squashed all the commits as well.